go/integTest/shared_commands_test.go

```diff
@@ -1687,6 +1688,237 @@ func (suite *GlideTestSuite) TestSPop_LastMember() {
 	})
 }
 
+func (suite *GlideTestSuite) TestSMove() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key1 := "{key}-1-" + uuid.NewString()
+		key2 := "{key}-2-" + uuid.NewString()
+		key3 := "{key}-3-" + uuid.NewString()
+		stringKey := "{key}-4-" + uuid.NewString()
+		nonExistingKey := "{key}-5-" + uuid.NewString()
+		memberArray1 := []string{"1", "2", "3"}
+		memberArray2 := []string{"2", "3"}
+		t := suite.T()
+
+		res1, err := client.SAdd(key1, memberArray1)
+		assert.NoError(t, err)
+		assert.Equal(t, int64(3), res1.Value())
+
+		res2, err := client.SAdd(key2, memberArray2)
+		assert.NoError(t, err)
+		assert.Equal(t, int64(2), res2.Value())
+
+		// move an element
+		res3, err := client.SMove(key1, key2, "1")
+		assert.NoError(t, err)
+		assert.True(t, res3.Value())
+
+		res4, err := client.SMembers(key1)
+		assert.NoError(t, err)
+		assert.Len(t, res4, 2)
+		assert.Contains(t, res4, api.CreateStringResult("2"))
+		assert.Contains(t, res4, api.CreateStringResult("3"))
```

Comment:
Use `reflect.DeepEqual` to compare sets or arrays

Reply:
I took these `Contains` checks from other existing commands, I suppose its indeed not exactly a set equality check but the deepequal set comp logic would be even more bloated to do for each call to SMembers :/
```
expectedSet := map[api.Result[string]]struct{}{
	api.CreateStringResult("2"): struct{}{},
	api.CreateStringResult("3"): struct{}{},
}
assert.True(t, reflect.DeepEqual(expectedSet, res4))
```